messagequeue/messagequeue.go

```diff
@@ -0,0 +1,216 @@
+package messagequeue
```

Comment:
This isn't a generic message queue and is specific to bitswap and the directory and package name should reflect that.

Reply:
I was curious about that -- seems like all the packages inside bitswap are generically named -- for example, there's a message directory with a "message" package which seems extra generic.